pkg/authentication/authentication.go

```diff
@@ -0,0 +1,25 @@
+package authentication
+
+import (
+	"context"
+)
+
+const (
+	// SchemeBearer defines a Bearer token scheme.
+	SchemeBearer = "Bearer"
+)
+
+// Credentials contains information about an authentication request.
+type Credentials struct {
+	// Scheme contains information about the type of credentials being used.
+	Scheme string
+	// Token contains information the actual credential.
+	Token string
+}
+
+// Authentication contains a set of methods to authenticate users through different authentication providers such as
+// Auth0, PlatformIdentity, and such.
+type Authentication interface {
+	// VerifyCredentials verifies that the given credentials are valid and can be granted access.
```

Comment:
Authentication only limits itself to verify that an entity is who they claim they are (e.g. verify that the user is Alejo Carballude) or the entity is authorized to represent some other entity (e.g. verify that Github is contacting the service on behalf of Alejo Carballude. This is only possible if Alejo Carballude granted explicit access to Github to represent him.)

Access is verified at the authorization level. As an example, you may authenticate as a specific user by verifying your credentials, but not be granted access to the resource you're requesting.

```suggestion
	// VerifyCredentials verifies that the given credentials are valid.
	// This does not indicate that the user has access to the requested resource, and should
	// be verified in another step.
```

Reply:
Makes sense, I added `can be granted access` meaning that the user can access the system as a whole, without actually referring to permissions. We can certainly remove this line. Thanks!